spec/models/agents/twitter_retweet_agent_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+describe Agents::TwitterRetweetAgent do
+  before do
+    @options = { :expected_receive_period_in_days => '2' }
+    @retweet_agent = described_class.new(name: 'RetweetingFiend', options: @options)
+    @retweet_agent.service = services(:generic)
+    @retweet_agent.user = users(:bob)
+    @retweet_agent.save!
+
+    @event1 = Event.new
+    @event1.agent = agents(:bob_twitter_user_agent)
+    @event1.payload = { id: 123, text: 'So awesome.. gotta retweet' }
+    @event1.save!
+
+    @event2 = Event.new
+    @event2.agent = agents(:bob_twitter_user_agent)
+    @event2.payload = { id: 456, text: 'Something Justin Bieber said' }
+    @event2.save!
+  end
+
+  describe '#receive' do
+    before do
+      @tweet1 = Twitter::Tweet.new(
+        id: @event1.payload[:id],
+        text: @event1.payload[:text]
+      )
+      @tweet2 = Twitter::Tweet.new(
+        id: @event2.payload[:id],
+        text: @event2.payload[:text]
+      )
+    end
+
+    context 'when the twitter client succeeds retweeting' do
+      context 'single incoming event' do
+        it 'should retweet the tweet from the payload' do
+          mock(@retweet_agent.twitter).retweet([@tweet1])
```

Comment:
This and the other examples below could easily be changed to a `mock.any_instance_of` if the caching of `twitter` is a bad idea.